### ISSUE TITLE
Is it possible to install only native windows/linux/mac version of Closure Compiler, and skip Java version altogether?

### ISSUE THREAD

**Issue by juj** [google/closure-compiler-npm#228]
I find that when I perform a `npm install` of `google-closure-compiler`, I always get `google-closure-compiler-java` installed. That installation takes up 12.5MB of extra space on disk, and I would like to avoid this, and instead always use one of the native versions instead.

Is this kind of npm installation possible somehow?

**Comment by frigus02** [comment]
> Would that be a safe thing to do?
I think that should work fine at the moment. Though we can't guarantee this will continue to work for future versions.

> Would it be possible for Closure to officially support "Java-free" installation?
It sounds possible in theory. Though to set expectations: we don't have plans to work on this anytime soon.

**Comment by ChadKillingsworth** [comment]
With the way npm installs work and the fact that the native images use optional packages to select the correct OS, we have to have one required dependency with the compiler runtime. Java is that required dependency as it is not OS specific.

Could we change the package code to be more tolerant of the java package being missing? Possibly, but when some of the native images are 200+ Mb, it hardly seems worth the effort for 12.5 Mb.

In short, I just don't see this work as a priority for any maintainer. We would be happy to evaluate a PR for this, but it would have to make sense for the broader ecosystem.

I'm going to close this issue for now as the current state is truly working as designed. A community PR is still an option though.

**Comment by juj** [comment]
I got back to looking at this area, and it was commented that

> when some of the native images are 200+ Mb, it hardly seems worth the effort for 12.5 Mb.

looking at our Emscripten packager on Windows, the sizes that I am seeing are:
```
13,785  node_modules\google-closure-compiler
13,339  node_modules\google-closure-compiler-java
41,257  node_modules\google-closure-compiler-windows
```
so deleting the java version saved up -19.5% from the total distribution size. It is definitely not nothing for our CDN costs.

In Emscripten I ended up running a post-installer batch script to search-replace according to https://github.com/google/closure-compiler-npm/issues/228#issuecomment-1063188952 , so as long as that invariant will keep holding, we would be happy.
